File: core/src/main/java/fr/sncf/osrd/stdcm/BacktrackingEnvelopeAttr.java
package fr.sncf.osrd.stdcm;

import fr.sncf.osrd.envelope.EnvelopeAttr;

public final class BacktrackingEnvelopeAttr implements EnvelopeAttr {
    @Override
    public Class<? extends EnvelopeAttr> getAttrType() {
        return BacktrackingEnvelopeAttr.class;
    }
}

File: core/src/main/kotlin/fr/sncf/osrd/stdcm/BacktrackingEnvelopeAttr.kt
package fr.sncf.osrd.stdcm

import fr.sncf.osrd.utils.SelfTypeHolder

class BacktrackingSelfTypeHolder : SelfTypeHolder {
    override val selfType: Class<out SelfTypeHolder>
        get() = BacktrackingSelfTypeHolder::class.java
}
